### PR TITLE
[feat] remove -t parameter and replace it with a check of whether it's tty

### PR DESCRIPTION
**Pull request by MangoIV**
I haven't tested the manual version yet...

### CODE REVIEW

**Comment by Mic92** [comment]
Looks good otherwise.

**Comment by MangoIV** [comment]
Yeah I’ll probably get the chance to try it out soon, I’ll report back then. :)

**Comment by MangoIV** [comment]
it works...

File: src/nixos-anywhere.sh [hunk 1]
```diff
@@ -180,6 +174,10 @@ if [[ ${print_build_logs-n} == "y" ]]; then
   nix_options+=("-L")
 fi
 
+if [ -t 0 ]; then
+  ssh_tty_param="-t"
+fi
```
Comment (zimbatm) [inline src/nixos-anywhere.sh:179]: Move this to line 71 so it becomes the detected default.

File: src/nixos-anywhere.sh [hunk 2]
```diff
@@ -106,9 +103,6 @@ while [[ $# -gt 0 ]]; do
     shift
     shift
     ;;
-  -t | --tty)
-    ssh_tty_param="-t"
-    ;;
```
Comment (zimbatm) [inline src/nixos-anywhere.sh:111]: This is useful for tools that might want to wrap up this program and interact with it as if they had a TTY.

Comment (MangoIV) [reply src/nixos-anywhere.sh:111]: I would leave it in if @Mic92 and @RaitoBezarius don't have anything against it.

Comment (Mic92) [reply src/nixos-anywhere.sh:111]: @zimbatm according to @Lassulus ssh blocks if the input is a pipe and not a tty.

Comment (zimbatm) [reply src/nixos-anywhere.sh:111]: Ok, I defer back to Lassulus

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**